src/otx/algorithms/common/adapters/torch/utils/utils.py

```diff
@@ -0,0 +1,48 @@
+"""Collections of util functions related to torch."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from torch.nn import Module
+
+try:
+    from timm.models.layers import convert_sync_batchnorm as timm_cvt_sycnbn
+except ImportError:
+    timm_cvt_sycnbn = None
+
+
+def model_from_timm(model: Module) -> bool:
+    """Check a model comes from timm module.
+
+    Args:
+        model (Module): model to check it comes from timm module.
+
+    Returns:
+        bool : whether model comes from timm or not.
+    """
+    if "timm" in model.__module__.split("."):
+        return True
+
+    is_fisrt = True
+    for sub_module in model.modules():
+        if is_fisrt:  # First module is the module itself.
+            is_fisrt = False
+            continue
```

Comment:
Nitpick: why don't you just iterate from 2nd item?

Reply:
Because model.modules() returns generator. I just want to keep memory saving.